R/utils_query_database.R

```diff
@@ -79,6 +79,21 @@ normalise_uuid <- function(.data, keep_uuid = FALSE) {
   .data
 }
 
+anonymize_uuid <- function(.data, keep_uuid = FALSE) {
+  # return(.data)
+  unique_uuid <- unique(.data$user_id)
+  data_count <- length(unique_uuid)
+  message(data_count)
+  anon_uuid = sapply(sample(1:data_count), function(.i) paste0("user_", .i))
```

Comment:
Note that, all users will get new uuids every time you refresh the database by pressing the `Reload data` button on the sidebar of the dashboard.

Reply:
Yes. That is a feature and not a bug. They really just want to see where people are traveling in general, not preserve the linkage between users and demographics.

Reply:
@asiripanich I didn't see that you had changed this to 

```
  unique_uuid <- unique(.data$user_id)
  anon_uuid <- paste0("user_", sample(length(unique_uuid)))
  .data$user_id = anon_uuid
```

That won't work because the trips table may have 200 entries from only 5 users.
You need to map each uuid to the corresponding anon uuid.

If you don't do that, you get the error

```
emdash_1           | Warning: Error in [[<-.data.frame: replacement has 5 rows, data has 12
```

like I just did 😄 

If you can think of a better way of doing this that is not a for loop, that is would be great.